### ISSUE TITLE
Memory leak when using numpy matrix in tests

### ISSUE THREAD

**Issue by lindera-steffen** [pytest-dev/pytest#10860]
Hi, when using non-empty numpy matrices in tests they will not be garbage collected and pile up slowly using more and more memory.

pip list output:
`attrs          22.2.0 
exceptiongroup 1.1.1  
iniconfig      2.0.0  
numpy          1.24.2 
packaging      23.0   
pip            10.0.1 
pluggy         1.0.0  
pytest         7.2.2  
setuptools     39.1.0 
tomli          2.0.1  `

Operating system:
Ubuntu 18.04.6 LTS
python 3.8

Minimal example:
`
import numpy as np

class TestMemory:
    def test_memory(self):
        for _ in range(1000000):
            test = np.matrix(np.random.rand(100, 100))
`

np.array does not share the same issue, so one can avoid matrix as much as possible, but for some operations it is still needed.

**Comment by Zac-HD** [comment]
I'd guess this is something about our traceback handling or assertion rewriting (if so `pytest --tb=native --assert=plain` should work).  Probably not a high priority though, since [`numpy` recommends against using `np.matrix`](https://numpy.org/doc/stable/reference/generated/numpy.matrix.html).

**Comment by lindera-steffen** [comment]
yeah, I am currently working on getting rid of matrices, but I need tests to check everything still works the same. So thanks for the command line values!
